### ISSUE TITLE
Improve DFtoVW user experience

### ISSUE THREAD

**Issue by etiennekintzler** [VowpalWabbit/vowpal_wabbit#2458]
I have been thinking (and playing) with the `DFtoVW` and I think **the class API could be improved**. Except for the convenience function `DFtoVW.from_colnames` which is quite clear (maybe except its name) the default `DFtoVW` constructor could be further simplified:

1. The user shouldn't have to use `Col` (although this class could be kept for internal use).

2. A parameter `features`, that build a naked namespace (no name nor scaling factor) with supplied `Feature` object(s), could be introduced.

I detail these two points below. If you think these two points are valid, I can of course contribute on this issue. Suggestions are welcomed !

-----

## 1. User shouldn't have to use `Col`

The main problem with `Col` is that it's tedious write and not really legible. For instance if we want to create the resulting line " | feat_x:x feat_y:y", (with dataframe df with two colmn a and b), the user has to write: 

```python
DFtoVW(namespaces=Namespace([Feature(Col("a"), "feat_a"), Feature(Col("b"), "feat_b")]))
```

I might be wrong here but **for most of the use case, the *value* of the feature is a column and the *name* of the feature is a constant**. If it is the case, then default behavior of `Feature` should reflect that. 

Instead of using `Col` we could just add two boolean parameters that specify whether the argument passed is a column or not. In doing so, we can: 
- handle the edge cases (name from a column and/or constant feature value) 
- keep the API simple for most of the usages.

The new signature of Feature would then be : 
```python
Feature(value, name=None, value_from_df=True, name_from_df=False)
```
Hence the current formulation:
```python
Feature(Col("a"), "feat_a")
Feature("a", "feat_a")
Feature("a", Col("b"))
```
would become:
```python
Feature("a", "feat_a")
Feature("a", "feat_a", value_from_df=False)
Feature("a", "b", name_from_df=True)
```

## 2. Parameter `features` in the constructor

To my knowledge, Namespace is a term highly specific to Vowpal Wabbit input format. It is not found elsewhere in the other python ML packages that the user might use (scikit-learn, statsmodels, keras, pytorch or tensorflow). Since multiple namespaces are for advanced usage, I suggest to create a parameter `features` (along with `namespaces`) that build a naked namespace.

In the current API formulation, if the user want to create the " | x" (for prediction purpose) he has to write (for column "a" of dataframe "df"):
```python
DFtoVW(namespaces=Namespace(Feature(Col("a"))))
```
which is quite awkward. With the `features ` parameter (and then new `Feature` class introduced in the 1st part) it would become:
```python
DFtoVW(features=Feature("a"))
```

Another example : 
```python
 DFtoVW(df=df, 
        label=SimpleLabel(Col("y")), 
        namespaces=Namespace([Feature(Col("a")), Feature(Col("b")), Feature("c")]))
```
would become 
```python
DFtoVW(df=df, label=SimpleLabel("y"), features=[Feature("a"), Feature("b"), Feature("c")])
```

**Comment by jackgerrits** [comment]
Thanks for your continued thought and work!

I'm a little skeptical for 1. Even though you correctly point out the most common behavior, I feel as though the implicit treatment of different parameters may be confusing. I would be interested to hear other people perspectives on this.

I am very on board for 2. Having the initial complexity be lower for the common use case and allowing a power user to dive deeper is great.

**Comment by jackgerrits** [comment]
> I think feature value that depends on a column of the dataframe and fixed name is what people expect most of the time.

I agree.

I only really ever forsee people wanting to use constant names for feature names. Constants in other places are very unlikely

>  but I am pretty positive that this should be settle through parameters instead of exposing Col class to the end user.

I agree

Your responses all sound reasonable to me.

**Comment by etiennekintzler** [comment]
Ok cool ! 

I will do a new PR soon with these 2 improvements (along with multi class label and a more stringent type checking).
